src/ApplicationSampler.cpp

```diff
@@ -155,15 +155,8 @@ namespace geopm
         else {
             double time_delta = geopm_time_diff(&m_update_time, &curr_time);
             for (int cpu_idx = 0; cpu_idx != m_num_cpu; ++cpu_idx) {
-                try {
-                    m_hint_time[cpu_idx][m_hint_last[cpu_idx]] += time_delta;
-                    m_hint_last[cpu_idx] = m_status->get_hint(cpu_idx);
-                }
-                catch (const std::out_of_range& oor) {
-                    throw Exception("ApplicationSamplerImp::" + std::string(__func__) +
-                                    "(): Out-of-range error while updating hints:" + oor.what(),
-                                    GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
-                }
```

Comment:
Very minor note (not worth stopping the PR):

It might be worth replacing this with an invariant check prior to the loop: GEOPM_DEBUG_ASSERT that m_hint_time.size() == m_hint_last() == m_num_cpu. It would also serve as "code documentation" in a way.